src/NzbDrone.Core/Indexers/FileList/FileListSettings.cs

```diff
@@ -0,0 +1,70 @@
+using System.Collections.Generic;
+using FluentValidation;
+using NzbDrone.Core.Annotations;
+using NzbDrone.Core.Validation;
+
+namespace NzbDrone.Core.Indexers.FileList
+{
+    public class FileListSettingsValidator : AbstractValidator<FileListSettings>
+    {
+        public FileListSettingsValidator()
+        {
+            RuleFor(c => c.BaseUrl).ValidRootUrl();
+            RuleFor(c => c.Username).NotEmpty();
+            RuleFor(c => c.Passkey).NotEmpty();
+
+            RuleFor(c => c.SeedCriteria).SetValidator(_ => new SeedCriteriaSettingsValidator());
+        }
+    }
+
+    public class FileListSettings : ITorrentIndexerSettings
+    {
+        private static readonly FileListSettingsValidator Validator = new FileListSettingsValidator();
+
+        public FileListSettings()
+        {
+            BaseUrl = "https://filelist.io";
+            MinimumSeeders = IndexerDefaults.MINIMUM_SEEDERS;
+
+            Categories = new int[]
+            {
+                (int)FileListCategories.AUDIO,
+                (int)FileListCategories.FLAC
+            };
+        }
+
+        [FieldDefinition(0, Label = "Username", Privacy = PrivacyLevel.UserName)]
+        public string Username { get; set; }
+
+        [FieldDefinition(1, Label = "Passkey", Privacy = PrivacyLevel.ApiKey)]
+        public string Passkey { get; set; }
+
+        [FieldDefinition(3, Label = "API URL", Advanced = true, HelpText = "Do not change this unless you know what you're doing. Since your API key will be sent to that host.")]
+        public string BaseUrl { get; set; }
+
+        [FieldDefinition(4, Label = "Categories", Type = FieldType.Select, SelectOptions = typeof(FileListCategories), HelpText = "Categories for use in search and feeds")]
+        public IEnumerable<int> Categories { get; set; }
+
+        [FieldDefinition(5, Type = FieldType.Number, Label = "Early Download Limit", Unit = "days", HelpText = "Time before release date Lidarr will download from this indexer, empty is no limit", Advanced = true)]
+        public int? EarlyReleaseLimit { get; set; }
+
+        [FieldDefinition(6, Type = FieldType.Number, Label = "Minimum Seeders", HelpText = "Minimum number of seeders required.", Advanced = true)]
+        public int MinimumSeeders { get; set; }
+
+        [FieldDefinition(7)]
+        public SeedCriteriaSettings SeedCriteria { get; set; } = new SeedCriteriaSettings();
+
+        public NzbDroneValidationResult Validate()
+        {
+            return new NzbDroneValidationResult(Validator.Validate(this));
+        }
+    }
+
+    public enum FileListCategories
+    {
+        [FieldOption]
+        AUDIO = 5,
+        [FieldOption]
+        FLAC = 11
```

Comment:
Leave audio and add ebook catecory: DOCS = 16

Reply:
Commited